src/REPLMode.jl

```diff
@@ -502,64 +576,106 @@ function enforce_argument(raw_args::Vector{String}, spec::ArgSpec)::PkgArguments
-    end
-    return opt
-end
-
-function enforce_meta_options(options::Vector{String}, specs::Dict{String,OptionSpec})::Vector{Option}
-    meta_opt_names = keys(specs)
-    return map(options) do opt
-        tok = enforce_option(opt, specs)
-        tok.val in meta_opt_names ||
-            pkgerror("option '$opt' is not a valid meta option.")
-            #TODO hint that maybe they intended to use it as a command option
-        return tok
+        if option.argument === nothing
+            repl_error(ERROR_OPT_NO_ARG, option)
+        end
     end
 end
 
-function enforce_opts(options::Vector{String}, specs::Dict{String,OptionSpec})::Vector{Option}
+# checking relationships between options
+function enforce_option(options::Vector{Option}, specs::Dict{String,OptionSpec})
     unique_keys = Symbol[]
     get_key(opt::Option) = specs[opt.val].api.first
 
-    # final parsing
-    toks = map(x->enforce_option(x,specs),options)
-    # checking
-    for opt in toks
-        # valid option
-        opt.val in keys(specs) ||
-            pkgerror("option '$(opt.val)' is not supported")
-        # conflicting options
-        key = get_key(opt)
+    foreach(x->enforce_option(x,specs), options)
+    # conflicting options
+    for option in options
+        key = get_key(option)
         if key in unique_keys
-            conflicting = filter(opt->get_key(opt) == key, toks)
-            pkgerror("Conflicting options: $conflicting")
-        else
-            push!(unique_keys, key)
+            conflicting = filter(opt->get_key(opt) == key, options)
+            repl_error(ERROR_CONFLICTING_KEYS, conflicting)
         end
+        push!(unique_keys, key)
     end
-    return toks
 end
 
-# this the entry point for the majority of input checks
+# this the entry point for the majority of "semantic" input checks
 function PkgCommand(statement::Statement)::PkgCommand
-    meta_opts = enforce_meta_options(statement.meta_options,
-                                     meta_option_specs)
-    args = enforce_argument(statement.arguments,
-                            statement.command.argument_spec)
-    opts = enforce_opts(statement.options, statement.command.option_specs)
-    return PkgCommand(meta_opts, statement.command, opts, args)
-end
+    cmd(statement::Statement) = "`$(statement.command_name)`"
+    ismeta=true
+    try
+        enforce_option(statement.meta_options, meta_option_specs)
+        ismeta=false
+        args = enforce_argument(statement.arguments, statement.command.argument_spec)
+        enforce_option(statement.options, statement.command.option_specs)
+        return PkgCommand(statement.meta_options, statement.command, statement.options, args)
+    catch ex
+        (ex isa PkgError && ex.class == PKG_ERROR_REPL) || rethrow()
+        if ex.code == ERROR_CONFLICTING_KEYS
+            opts = join(map(wrap_option, ex.state), ", ")
+            if ismeta
+                ex.msg = "$opts are conflicting meta options"
```

Comment:
Yeah, I'll just cut out all of that "meta option" stuff. It was in place because I assumed there would be more options like `--env` but its pretty useless now that `--env` is gone.